### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by MarijaRomanenkova**

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...xpression","React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array.",["173"],"jsx-a11y/anchor-is-valid","The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md","JSXOpeningElement",["174"],{"desc":"175","fix":"176"},{"desc":"177","fix":"178"},{"desc":"179","fix":"180"},{"kind":"181","justification":"182"},"Update the dependencies array to be: [currentCategoryOptionsByLanguage, currentLevel, findCurrentLevelOptionsByIndex]",{"range":"183","text":"184"},"Update the dependencies array to be: [correctAnswerID, thisCategoryOptionsByLanguage]",{"range":"185","text":"186"},"Update the dependencies array to be: [dispatch]",{"range":"187","text":"188"},"directive","",[2456,2504],"[currentCategoryOptionsByLanguage, currentLevel, findCurrentLevelOptionsByIndex]",[3063,3094],"[correctAnswerID, thisCategoryOptionsByLanguage]",[483,485],"[dispatch]"]
```
Comment (viktarrudzenia) [inline .eslintcache:1]: do we really need it?

File: .eslintrs.cjs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+module.exports = {
```
Comment (viktarrudzenia) [inline .eslintrs.cjs:1]: why does this file have .cjs on end instead .js?

File: .eslintrs.cjs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    'valid-jsdoc': 'off',
+    'react/boolean-prop-naming': 'error',
+    'react/sort-prop-types': 'error',
+    'import/newline-after-import': ['error'],
+    'unused-imports/no-unused-imports': 'error',
+
+    // We need to disable "no-unused-vars" when working with TypeScript,
+    // but plugin:@typescript-eslint/recommended is already doing that for us.
+    // See https://typescript-eslint.io/rules/no-unused-vars/
+    '@typescript-eslint/no-unused-vars': 'error',
```
Comment (viktarrudzenia) [inline .eslintrs.cjs:38]: no, we need it, this rule is important

File: src/app/store.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-cycle
```
Comment (viktarrudzenia) [inline src/app/store.ts:2]: not sure to disable here import-no-cycle,
why did it happen?

File: src/features/game/correctAnswer/index.tsx [hunk 5]
```diff
@@ -26,34 +24,48 @@ interface AnswerToRender {
...
+  const correctAnswerID = useAppSelector(selectCorrectAnswerID);
+  const isCorrectAnswerChosen = useAppSelector(selectIsCorrectAnswerChosen);
+  const currentCategoryOptionsByLanguage = birdsData[language];
 
-  let currentCategoryOptions: AnswerOptionsArray | [] = [];
-  let correctAnswerObject: Option | {} = {};
-  function findCurrentLevelByIndex(option: Option, index: number): boolean {
-    return index === currentLevel;
-  }
+  const findCurrentLevelByIndex = (category: any, index: number): boolean =>
```
Comment (viktarrudzenia) [inline src/features/game/correctAnswer/index.tsx:36]: do we really need a `category` here?

File: src/features/game/gameSlice.ts [hunk 6]
```diff
@@ -1,14 +1,15 @@
-import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
-import { RootState, AppThunk } from 'app/store';
+/* eslint-disable import/no-cycle */
```
Comment (viktarrudzenia) [inline src/features/game/gameSlice.ts:1]: this one should gone

File: src/shared/ui/button/index.tsx [hunk 7]
```diff
@@ -1,26 +1,26 @@
...
 
 import styles from './index.module.scss';
 
 interface ButtonProps {
   isDisabled: boolean;
   handleClick: MouseEventHandler;
-  name: any;
-  styles?: { [key: string]: string };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  text: any;
```
Comment (viktarrudzenia) [inline src/shared/ui/button/index.tsx:10]: why text any here?
looks like it will be always string

**Review by viktarrudzenia — CHANGES_REQUESTED** [review]
looks ok, just need to return important eslint rules

File: .eslintcache (hunk 1 above)
Comment (MarijaRomanenkova) [reply .eslintcache:1]: fixed

Comment (MarijaRomanenkova) [reply .eslintcache:1]: No. fixed.

File: src/app/store.ts (hunk 4 above)
Comment (MarijaRomanenkova) [reply src/app/store.ts:2]: fixed

File: src/features/game/correctAnswer/index.tsx (hunk 5 above)
Comment (MarijaRomanenkova) [reply src/features/game/correctAnswer/index.tsx:36]: fixed

File: src/features/game/gameSlice.ts (hunk 6 above)
Comment (MarijaRomanenkova) [reply src/features/game/gameSlice.ts:1]: fixed

File: src/shared/ui/button/index.tsx (hunk 7 above)
Comment (MarijaRomanenkova) [reply src/shared/ui/button/index.tsx:10]: you are right. fixed

File: .eslintrs.cjs (hunk 3 above)
Comment (MarijaRomanenkova) [reply .eslintrs.cjs]: Got it.

File: src/app/store.ts (hunk 4 above)
Comment (MarijaRomanenkova) [reply src/app/store.ts]: fixed

File: src/features/game/correctAnswer/index.tsx (hunk 5 above)
Comment (MarijaRomanenkova) [reply src/features/game/correctAnswer/index.tsx]: fixed

File: src/features/game/gameSlice.ts (hunk 6 above)
Comment (MarijaRomanenkova) [reply src/features/game/gameSlice.ts]: fixed

File: .eslintrs.cjs (hunk 2 above)
Comment (MarijaRomanenkova) [reply .eslintrs.cjs]: fixed

File: .eslintrs.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
+module.exports = {
```
Comment (viktarrudzenia) [inline .eslintrs.js:1]: image
is it working correctly?
it looks like it should be named as .esclintrc.js instead of .eslintrS.js

**Review by viktarrudzenia — APPROVED** [review]
code looks good, look into eslintrc, if it worked well - feel free to merge using squash merge

**Pull request closed without merging**